### ISSUE TITLE
"UnicodeDecodeError" in 'fileserver.file_list()'

### ISSUE THREAD

**Issue by eliasp** [saltstack/salt#20499]
Master on Ubuntu 14.04.1:
```
               Salt: 2014.7.1
             Python: 2.7.6 (default, Mar 22 2014, 22:59:56)
             Jinja2: 2.7.2
           M2Crypto: 0.21.1
     msgpack-python: 0.3.0
       msgpack-pure: Not Installed
           pycrypto: 2.6.1
            libnacl: Not Installed
             PyYAML: 3.10
              ioflo: Not Installed
              PyZMQ: 14.0.1
               RAET: Not Installed
                ZMQ: 4.0.4
               Mako: 0.9.1

```

I'm continuously getting this backtrace in the master's log:
```
2015-02-07 21:51:36,769 [salt.master      ][ERROR   ] Error in function _file_list:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/master.py", line 1336, in run_func
    ret = getattr(self, func)(load)
  File "/usr/lib/python2.7/dist-packages/salt/fileserver/__init__.py", line 455, in file_list
    return sorted(ret)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe4 in position 64: ordinal not in range(128)
```

The backtrace is triggered by `sorted(ret)` in `file_list()`.
The types of the elements in the set are `str` and `unicode`.

I might have a fix ready for this - just testing a bit and will submit a PR soon.

**Comment by eliasp** [comment]
Fix available in #20500

**Comment by eliasp** [comment]
#20500 was merged - closing!
